src/test/java/picard/sam/BamErrorMetric/BaseErrorAggregationTest.java

```diff
@@ -0,0 +1,192 @@
+package picard.sam.BamErrorMetric;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.reference.SamLocusAndReferenceIterator.SAMLocusAndReference;
+import htsjdk.samtools.util.SamLocusIterator;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by farjoun on 5/31/18.
+ */
+public class BaseErrorAggregationTest {
+
+    @Test
+    public void testBaseErrorAggregation() {
+        final SAMSequenceRecord samSequenceRecord = new SAMSequenceRecord("chr1", 200);
+        final SAMFileHeader samFileHeader = new SAMFileHeader();
+        samFileHeader.addSequence(samSequenceRecord);
+
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+
+        final List<SAMRecord> samRecords = builder.addPair("Read1234", 0, 1, 1,
+                false, false, "16M", "16M", true, false, 20);
+        final SAMRecord samRecord1 = samRecords.get(0);
+        final SAMRecord samRecord2 = samRecords.get(1);
+
+        samRecord1.setReadBases("CgTGtGGAcAAAgAAA".getBytes());
+        samRecord2.setReadBases("CcTGGtGAcAAAgAAA".getBytes());
+        final byte[] refBases = "CATGGGGAAAAAAAAA".getBytes();
+
+        BaseErrorAggregation<?> baseErrorAggregation = new BaseErrorAggregation<>(OverlappingReadsErrorCalculator::new, ReadBaseStratification.readOrdinalityStratifier);
+        final int length = refBases.length;
+
+        for (int i = 0; i < length; i++) {
```

Comment:
It looks like this `for` loop pattern occurs many times in different tests. It would be better to replace the duplicate code with a common method.